android-components/components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/manager/FxaAccountManager.kt

```diff
@@ -418,6 +418,12 @@ open class FxaAccountManager(
      */
     suspend fun logout() = withContext(coroutineContext) { processQueue(Event.Account.Logout) }
 
+    /**
+     * Should be called once the user deletes the account.
+     * It clears the account data and notifies the observers that account has been deleted.
+     */
+    suspend fun onAccountDeleted() = withContext(coroutineContext) { processQueue(Event.Account.Delete) }
```

Comment:
Was debating on that myself, I don't like the current name because of the callback association, but I find `deleteAccount` to be a poor choice as well. `FxaAccountManager` doesn't delete the account (and can't), the deletion happened outside, and the manager reacts to the external event. Could we find a better naming? `handleAccountDeletion`, `processAccountDeletion`, `respondToAccountDeletion`?